### ISSUE TITLE
Using an interceptor or a guard on a "@ResolveProperty" method does not work as expected.

### ISSUE THREAD

**Issue by htondkar** [nestjs/graphql#322]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

when used like:

```ts
 @ResolveProperty('books', returns => [BookNode])
 @UseInterceptors(LogInterceptor(() => `books`))
 async books() {
    return this.bookService.getAllBooks();
 }
```

the intercept method on the interceptor is never called although the use-case is valid.

but this works:

```ts
  @Query(returns => ViewerNode)
  @UseInterceptors(LogInterceptor(() => `viewer`))
  async viewer() {
    return createViewer();
  }
```

seems like "UseInterceptors" does not work with ResolveProperty decorator.

## Expected behavior

Using the "UseInterceptors" or "UseGuards" on methods that are property revolvers should work like any other method (query, routes, etc).

## Minimal reproduction of the problem with instructions

You can find a full example here:
https://github.com/smmoosavi/nestjs-graphql-example
steps to reproduce:

1. clone and run the server
2. open the graphql playground and execute this query:

```gql
query {
  viewer {
    books {
      id
    }
  }
}
```

3. we expect to see two logs, one for the viewer and one of the books, but we see only the viewer.

## What is the motivation / use case for changing the behavior?

[this issue](https://github.com/nestjs/graphql/issues/238) resulted in removal of this feature but using these decorators on ResolveProperty makes complete sense, as we have demonstrated in the example above [in the repo](https://github.com/smmoosavi/nestjs-graphql-example)
In conclusion it seems that using these decorators on methods makes complete sense, but using them on a class might be problematic.

## Environment

<pre><code>
Nest version: 
├─ @nestjs/common@6.4.1
├─ @nestjs/core@6.4.1
├─ @nestjs/graphql@6.2.4
├─ @nestjs/platform-express@6.4.1
├─ @nestjs/testing@6.4.1
├─ type-graphql@0.17.4

 
For Tooling issues:
- Node version: 10.16.0  
- Platform: Linux 

Others:

</code></pre>

**Comment by kamilmysliwiec** [comment]
See https://github.com/nestjs/graphql/issues/295
